### ISSUE TITLE
Car example - does not converge

### ISSUE THREAD

**Issue by adamlofts** [MorvanZhou/Reinforcement-learning-with-tensorflow#4]
I've been experimenting with your car example. The success of the training seems to depend a lot on when the training stops.

See the log below as an example. It reaches a good state of 600 steps by EP 210 and the car is going around the block. If the training stops now then the model works well when `LOAD = True`. But at EP 232 the car is crashing again. So if we happen to stop training on EP 232 we will not get a successful model saved.

If you leave the training running you will continually see the steps reach 600 and then drop back down and build up again. Is this behaviour expected? Is there a regularisation issue?

```
Ep: 0 | Steps: 34 | Explore: 2.00
Ep: 1 | Steps: 50 | Explore: 2.00
Ep: 2 | Steps: 57 | Explore: 2.00
Ep: 3 | Steps: 17 | Explore: 2.00
Ep: 4 | Steps: 56 | Explore: 2.00
Ep: 5 | Steps: 59 | Explore: 2.00
Ep: 6 | Steps: 59 | Explore: 2.00
Ep: 7 | Steps: 21 | Explore: 2.00
Ep: 8 | Steps: 61 | Explore: 2.00
Ep: 9 | Steps: 58 | Explore: 2.00
Ep: 10 | Steps: 41 | Explore: 2.00
Ep: 11 | Steps: 26 | Explore: 2.00
Ep: 12 | Steps: 58 | Explore: 2.00
Ep: 13 | Steps: 58 | Explore: 2.00
Ep: 14 | Steps: 29 | Explore: 2.00
Ep: 15 | Steps: 60 | Explore: 2.00
Ep: 16 | Steps: 59 | Explore: 2.00
Ep: 17 | Steps: 59 | Explore: 2.00
Ep: 18 | Steps: 54 | Explore: 2.00
Ep: 19 | Steps: 57 | Explore: 2.00
Ep: 20 | Steps: 62 | Explore: 2.00
Ep: 21 | Steps: 57 | Explore: 2.00
Ep: 22 | Steps: 58 | Explore: 2.00
Ep: 23 | Steps: 31 | Explore: 2.00
Ep: 24 | Steps: 59 | Explore: 2.00
Ep: 25 | Steps: 58 | Explore: 2.00
Ep: 26 | Steps: 32 | Explore: 2.00
Ep: 27 | Steps: 58 | Explore: 2.00
Ep: 28 | Steps: 41 | Explore: 2.00
Ep: 29 | Steps: 30 | Explore: 2.00
Ep: 30 | Steps: 40 | Explore: 2.00
Ep: 31 | Steps: 39 | Explore: 2.00
Ep: 32 | Steps: 57 | Explore: 2.00
Ep: 33 | Steps: 26 | Explore: 2.00
Ep: 34 | Steps: 29 | Explore: 2.00
Ep: 35 | Steps: 59 | Explore: 2.00
Ep: 36 | Steps: 59 | Explore: 2.00
Ep: 37 | Steps: 37 | Explore: 2.00
Ep: 38 | Steps: 59 | Explore: 2.00
Ep: 39 | Steps: 20 | Explore: 2.00
Ep: 40 | Steps: 36 | Explore: 2.00
Ep: 41 | Steps: 144 | Explore: 2.00
Ep: 42 | Steps: 37 | Explore: 2.00
Ep: 43 | Steps: 57 | Explore: 2.00
Ep: 44 | Steps: 61 | Explore: 2.00
Ep: 45 | Steps: 58 | Explore: 2.00
Ep: 46 | Steps: 41 | Explore: 2.00
Ep: 47 | Steps: 61 | Explore: 2.00
Ep: 48 | Steps: 53 | Explore: 2.00
Ep: 49 | Steps: 60 | Explore: 2.00
Ep: 50 | Steps: 24 | Explore: 2.00
Ep: 51 | Steps: 58 | Explore: 2.00
Ep: 52 | Steps: 58 | Explore: 2.00
Ep: 53 | Steps: 24 | Explore: 2.00
Ep: 54 | Steps: 39 | Explore: 2.00
Ep: 55 | Steps: 58 | Explore: 2.00
Ep: 56 | Steps: 23 | Explore: 2.00
Ep: 57 | Steps: 31 | Explore: 2.00
Ep: 58 | Steps: 65 | Explore: 2.00
Ep: 59 | Steps: 37 | Explore: 2.00
Ep: 60 | Steps: 24 | Explore: 2.00
Ep: 61 | Steps: 33 | Explore: 2.00
Ep: 62 | Steps: 14 | Explore: 2.00
Ep: 63 | Steps: 57 | Explore: 2.00
Ep: 64 | Steps: 60 | Explore: 2.00
Ep: 65 | Steps: 58 | Explore: 2.00
Ep: 66 | Steps: 21 | Explore: 2.00
Ep: 67 | Steps: 59 | Explore: 2.00
Ep: 68 | Steps: 20 | Explore: 2.00
Ep: 69 | Steps: 20 | Explore: 2.00
Ep: 70 | Steps: 64 | Explore: 2.00
Ep: 71 | Steps: 135 | Explore: 2.00
Ep: 72 | Steps: 21 | Explore: 2.00
Ep: 73 | Steps: 34 | Explore: 2.00
Ep: 74 | Steps: 33 | Explore: 2.00
Ep: 75 | Steps: 65 | Explore: 2.00
Ep: 76 | Steps: 58 | Explore: 2.00
Ep: 77 | Steps: 57 | Explore: 2.00
Ep: 78 | Steps: 22 | Explore: 2.00
Ep: 79 | Steps: 24 | Explore: 2.00
Ep: 80 | Steps: 59 | Explore: 2.00
Ep: 81 | Steps: 31 | Explore: 2.00
Ep: 82 | Steps: 31 | Explore: 2.00
Ep: 83 | Steps: 21 | Explore: 2.00
Ep: 84 | Steps: 57 | Explore: 2.00
Ep: 85 | Steps: 28 | Explore: 2.00
Ep: 86 | Steps: 21 | Explore: 2.00
Ep: 87 | Steps: 43 | Explore: 2.00
Ep: 88 | Steps: 19 | Explore: 2.00
Ep: 89 | Steps: 60 | Explore: 2.00
Ep: 90 | Steps: 59 | Explore: 2.00
Ep: 91 | Steps: 41 | Explore: 2.00
Ep: 92 | Steps: 53 | Explore: 2.00
Ep: 93 | Steps: 75 | Explore: 2.00
Ep: 94 | Steps: 55 | Explore: 2.00
Ep: 95 | Steps: 58 | Explore: 2.00
Ep: 96 | Steps: 50 | Explore: 2.00
Ep: 97 | Steps: 58 | Explore: 2.00
Ep: 98 | Steps: 44 | Explore: 2.00
Ep: 99 | Steps: 21 | Explore: 2.00
Ep: 100 | Steps: 63 | Explore: 2.00
Ep: 101 | Steps: 33 | Explore: 2.00
Ep: 102 | Steps: 27 | Explore: 2.00
Ep: 103 | Steps: 24 | Explore: 2.00
Ep: 104 | Steps: 58 | Explore: 2.00
Ep: 105 | Steps: 59 | Explore: 2.00
Ep: 106 | Steps: 32 | Explore: 2.00
Ep: 107 | Steps: 29 | Explore: 2.00
Ep: 108 | Steps: 82 | Explore: 1.92
Ep: 109 | Steps: 60 | Explore: 1.87
Ep: 110 | Steps: 25 | Explore: 1.84
Ep: 111 | Steps: 59 | Explore: 1.79
Ep: 112 | Steps: 41 | Explore: 1.75
Ep: 113 | Steps: 22 | Explore: 1.73
Ep: 114 | Steps: 36 | Explore: 1.70
Ep: 115 | Steps: 83 | Explore: 1.63
Ep: 116 | Steps: 19 | Explore: 1.62
Ep: 117 | Steps: 23 | Explore: 1.60
Ep: 118 | Steps: 30 | Explore: 1.58
Ep: 119 | Steps: 21 | Explore: 1.56
Ep: 120 | Steps: 29 | Explore: 1.54
Ep: 121 | Steps: 26 | Explore: 1.52
Ep: 122 | Steps: 31 | Explore: 1.49
Ep: 123 | Steps: 32 | Explore: 1.47
Ep: 124 | Steps: 92 | Explore: 1.40
Ep: 125 | Steps: 20 | Explore: 1.39
Ep: 126 | Steps: 21 | Explore: 1.38
Ep: 127 | Steps: 28 | Explore: 1.36
Ep: 128 | Steps: 23 | Explore: 1.34
Ep: 129 | Steps: 21 | Explore: 1.33
Ep: 130 | Steps: 43 | Explore: 1.30
Ep: 131 | Steps: 24 | Explore: 1.28
Ep: 132 | Steps: 19 | Explore: 1.27
Ep: 133 | Steps: 19 | Explore: 1.26
Ep: 134 | Steps: 36 | Explore: 1.24
Ep: 135 | Steps: 29 | Explore: 1.22
Ep: 136 | Steps: 26 | Explore: 1.20
Ep: 137 | Steps: 22 | Explore: 1.19
Ep: 138 | Steps: 30 | Explore: 1.17
Ep: 139 | Steps: 27 | Explore: 1.16
Ep: 140 | Steps: 52 | Explore: 1.13
Ep: 141 | Steps: 23 | Explore: 1.11
Ep: 142 | Steps: 29 | Explore: 1.10
Ep: 143 | Steps: 19 | Explore: 1.09
Ep: 144 | Steps: 22 | Explore: 1.08
Ep: 145 | Steps: 19 | Explore: 1.07
Ep: 146 | Steps: 20 | Explore: 1.05
Ep: 147 | Steps: 23 | Explore: 1.04
Ep: 148 | Steps: 72 | Explore: 1.01
Ep: 149 | Steps: 30 | Explore: 0.99
Ep: 150 | Steps: 21 | Explore: 0.98
Ep: 151 | Steps: 18 | Explore: 0.97
Ep: 152 | Steps: 35 | Explore: 0.95
Ep: 153 | Steps: 21 | Explore: 0.94
Ep: 154 | Steps: 28 | Explore: 0.93
Ep: 155 | Steps: 40 | Explore: 0.91
Ep: 156 | Steps: 47 | Explore: 0.89
Ep: 157 | Steps: 22 | Explore: 0.88
Ep: 158 | Steps: 21 | Explore: 0.87
Ep: 159 | Steps: 64 | Explore: 0.85
Ep: 160 | Steps: 39 | Explore: 0.83
Ep: 161 | Steps: 66 | Explore: 0.80
Ep: 162 | Steps: 34 | Explore: 0.79
Ep: 163 | Steps: 33 | Explore: 0.78
Ep: 164 | Steps: 59 | Explore: 0.75
Ep: 165 | Steps: 57 | Explore: 0.73
Ep: 166 | Steps: 58 | Explore: 0.71
Ep: 167 | Steps: 36 | Explore: 0.70
Ep: 168 | Steps: 28 | Explore: 0.69
Ep: 169 | Steps: 46 | Explore: 0.67
Ep: 170 | Steps: 36 | Explore: 0.66
Ep: 171 | Steps: 58 | Explore: 0.64
Ep: 172 | Steps: 31 | Explore: 0.63
Ep: 173 | Steps: 22 | Explore: 0.63
Ep: 174 | Steps: 22 | Explore: 0.62
Ep: 175 | Steps: 24 | Explore: 0.61
Ep: 176 | Steps: 24 | Explore: 0.60
Ep: 177 | Steps: 26 | Explore: 0.60
Ep: 178 | Steps: 20 | Explore: 0.59
Ep: 179 | Steps: 21 | Explore: 0.58
Ep: 180 | Steps: 24 | Explore: 0.58
Ep: 181 | Steps: 24 | Explore: 0.57
Ep: 182 | Steps: 21 | Explore: 0.56
Ep: 183 | Steps: 19 | Explore: 0.56
Ep: 184 | Steps: 24 | Explore: 0.55
Ep: 185 | Steps: 20 | Explore: 0.55
Ep: 186 | Steps: 24 | Explore: 0.54
Ep: 187 | Steps: 15 | Explore: 0.54
Ep: 188 | Steps: 15 | Explore: 0.53
Ep: 189 | Steps: 17 | Explore: 0.53
Ep: 190 | Steps: 19 | Explore: 0.52
Ep: 191 | Steps: 21 | Explore: 0.52
Ep: 192 | Steps: 17 | Explore: 0.51
Ep: 193 | Steps: 19 | Explore: 0.51
Ep: 194 | Steps: 18 | Explore: 0.50
Ep: 195 | Steps: 20 | Explore: 0.50
Ep: 196 | Steps: 23 | Explore: 0.49
Ep: 197 | Steps: 20 | Explore: 0.49
Ep: 198 | Steps: 25 | Explore: 0.48
Ep: 199 | Steps: 20 | Explore: 0.48
Ep: 200 | Steps: 22 | Explore: 0.47
Ep: 201 | Steps: 18 | Explore: 0.47
Ep: 202 | Steps: 20 | Explore: 0.46
Ep: 203 | Steps: 27 | Explore: 0.46
Ep: 204 | Steps: 19 | Explore: 0.45
Ep: 205 | Steps: 21 | Explore: 0.45
Ep: 206 | Steps: 24 | Explore: 0.44
Ep: 207 | Steps: 62 | Explore: 0.43
Ep: 208 | Steps: 61 | Explore: 0.42
Ep: 209 | Steps: 62 | Explore: 0.40
Ep: 210 | Steps: 600 | Explore: 0.30
Ep: 211 | Steps: 600 | Explore: 0.22
Ep: 212 | Steps: 600 | Explore: 0.16
Ep: 213 | Steps: 600 | Explore: 0.12
Ep: 214 | Steps: 600 | Explore: 0.10
Ep: 215 | Steps: 600 | Explore: 0.10
Ep: 216 | Steps: 600 | Explore: 0.10
Ep: 217 | Steps: 600 | Explore: 0.10
Ep: 218 | Steps: 600 | Explore: 0.10
Ep: 219 | Steps: 600 | Explore: 0.10
Ep: 220 | Steps: 600 | Explore: 0.10
Ep: 221 | Steps: 600 | Explore: 0.10
Ep: 222 | Steps: 600 | Explore: 0.10
Ep: 223 | Steps: 600 | Explore: 0.10
Ep: 224 | Steps: 600 | Explore: 0.10
Ep: 225 | Steps: 600 | Explore: 0.10
Ep: 226 | Steps: 600 | Explore: 0.10
Ep: 227 | Steps: 600 | Explore: 0.10
Ep: 228 | Steps: 600 | Explore: 0.10
Ep: 229 | Steps: 600 | Explore: 0.10
Ep: 230 | Steps: 600 | Explore: 0.10
Ep: 231 | Steps: 453 | Explore: 0.10
Ep: 232 | Steps: 30 | Explore: 0.10
Ep: 233 | Steps: 31 | Explore: 0.10
Ep: 234 | Steps: 95 | Explore: 0.10
Ep: 235 | Steps: 519 | Explore: 0.10
Ep: 236 | Steps: 33 | Explore: 0.10
Ep: 237 | Steps: 600 | Explore: 0.10
Ep: 238 | Steps: 163 | Explore: 0.10
```

**Comment by MorvanZhou** [comment]
Hi @adamlofts , that is indeed one problem. Do you try lower the learning rate? Cause the lr in the script is high as I want to see the impact quickly.
